File: client/src/app/_services/moj-elektro-api.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class MojElektroApiService {
  private http = inject(HttpClient);
  baseUrl = environment.apiUrl;

  getMeterReadings(usagePoint: string, startTime: string, endTime: string, option: string, apiToken: string) {
    const params = new HttpParams()
      .set('usagePoint', usagePoint)
      .set('startTime', startTime)
      .set('endTime', endTime)
      .set('option', option);
    return this.http.get<any>(
      `${this.baseUrl}mojelektroexternaldata/meter-readings`,
      {
        params,
        headers: { 'X-API-TOKEN': apiToken }
      }
    );
  }
}

File: API/Services/MojElektroApiService.cs
using System.Net.Http;
using System.Threading.Tasks;
using API.DTOs;
using Microsoft.Extensions.Configuration;
using Newtonsoft.Json;

namespace API.Services;

public class MojElektroApiService
{
    private readonly HttpClient _httpClient;
    private readonly string _baseUrl;

    public MojElektroApiService(HttpClient httpClient, IConfiguration config)
    {
        _httpClient = httpClient;
        _baseUrl = "https://mojelektro.informatika.si/api"; // or from config
    }

    public async Task<IntervalBlock> GetMeterReadingsAsync(string usagePoint, string startTime, string endTime, string option, string apiToken)
    {
        var url = $"https://api-test.informatika.si/mojelektro/v1/meter-readings?usagePoint={Uri.EscapeDataString(usagePoint)}&startTime={Uri.EscapeDataString(startTime)}&endTime={Uri.EscapeDataString(endTime)}&option={Uri.EscapeDataString(option)}";
        var request = new HttpRequestMessage(HttpMethod.Get, url);
        request.Headers.Add("accept", "application/json");
        request.Headers.Add("X-API-TOKEN", apiToken);
        var response = await _httpClient.SendAsync(request);
        response.EnsureSuccessStatusCode();
        if (!response.IsSuccessStatusCode)
        {
            throw new HttpRequestException($"Error fetching data: {response.ReasonPhrase}");
        }
        IntervalBlock? myDeserializedClass =
            JsonConvert.DeserializeObject<IntervalBlock>(await response.Content.ReadAsStringAsync());

        if (myDeserializedClass == null)
        {
            throw new JsonException("Failed to deserialize meter readings response.");
        }

        return myDeserializedClass;
    }
}
